### PR TITLE
watermark

### PR DESCRIPTION
**Pull request by pldespaigne**
- [x] generate a watermark in the backend
- [x] display the watermark in the player

### CODE REVIEW

File: apps/backend-functions/src/internals/resize-images.ts [hunk 1]
```diff
@@ -18,6 +18,11 @@ export async function onFileUploadEvent(data: functions.storage.ObjectMetadata)
     return false;
   }
 
+  if (data.contentType === 'image/svg+xml') {
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/internals/resize-images.ts:21]: Add a comment to explain why we shouldn't have that.
What about the other contentType (text, script, ...) ?

File: apps/backend-functions/src/internals/watermark.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { admin, db, getStorageBucketName } from './firebase';
+
+/**
+ * - Generate a svg file with the name & email of the user
+ * - Store the watermark file in the storage bucket
+ * - Update the user's firestore doc to link it to the watermark file
+ */
+export async function upsertWatermark(user: PublicUser) {
+
+  if (!user.firstName || !user.lastName) {
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/internals/watermark.ts:12]: You don't check for the email ?

File: apps/backend-functions/src/internals/watermark.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  const ref = `watermark/${user.uid}.svg`;
+  const file = admin.storage().bucket(getStorageBucketName()).file(ref);
+
+  await new Promise(res => {
+    file.createWriteStream({contentType: 'image/svg+xml'}).end(watermark, () => res());
+  });
+
+  const signedUrl = await file.getSignedUrl({action: 'read', expires: '01-01-3000', version: 'v2'});
+
+  user.watermark = {
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/internals/watermark.ts:27]: Be carefull the ImgRef model will evolve wih @ioaNikas's PR on multile size image.

File: apps/backend-functions/src/templates/watermark.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+
+export function getWatermark(firstName: string, lastName: string, email: string) {
+  return `
+    <svg viewBox="0 0 640 360" xmlns="http://www.w3.org/2000/svg">
+      <style>
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/templates/watermark.ts:5]: I'm not sure, but does this only apply on the current svg scope or is going to apply everywhere ?

File: apps/backend-functions/src/users.ts [hunk 5]
```diff
@@ -77,17 +78,43 @@ export const onUserCreate = async (user: UserRecord) => {
   // update Algolia index
   const userSnap = await userDocRef.get();
   const userData = userSnap.data() as PublicUser;
-  return storeSearchableUser(userData);
+
+  const promises: Promise<any>[] = [];
+  promises.push(storeSearchableUser(userData));
+  promises.push(upsertWatermark(userData));
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/users.ts:84]: Ok. Just an alternative here
```typescript
return Promise.all([
  storeSearchableUser(userData),
  upsertWatermark(userData)
])
```

File: apps/backend-functions/src/users.ts [hunk 6]
```diff
@@ -77,17 +78,43 @@ export const onUserCreate = async (user: UserRecord) => {
...
+  }
+
+  // if name or email has changed : update watermark
+  if (
+    before.firstName !== after.firstName ||
+    before.lastName !== after.lastName ||
+    before.email !== after.email
+  ) {
+    promises.push(upsertWatermark(after));
+  }
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/users.ts:115]: Better : 
```typescript
const changed = (keys: string[]) => keys.some(key => before[key] !== after[key]);
if (changed['firstName', 'lastName', 'email']) {
  if (before.avatar?.url !== after.avatar?.url) {
    promises.push(storeSearchableUser(after));
  }
  promises.push(upsertWatermark(after));
}
```

File: libs/event/src/lib/components/player/player.component.ts [hunk 7]
```diff
@@ -1,14 +1,16 @@
...
 import { DOCUMENT } from '@angular/common';
+import { AuthQuery } from '@blockframes/auth/+state';
 
 declare const jwplayer: any;
 
 @Component({
   selector: 'event-player',
   templateUrl: './player.component.html',
   styleUrls: ['./player.component.scss'],
+  encapsulation: ViewEncapsulation.None,
```
Comment (GrandSchtroumpf) [inline libs/event/src/lib/components/player/player.component.ts:13]: Add comment to explain why

File: apps/backend-functions/src/internals/watermark.ts (hunk 2 above)
Comment (pldespaigne) [reply apps/backend-functions/src/internals/watermark.ts:12]: email is mandatory in `PublicUser` where `firstName` & `lastName` are not

File: apps/backend-functions/src/internals/resize-images.ts (hunk 1 above)
Comment (pldespaigne) [reply apps/backend-functions/src/internals/resize-images.ts:21]: other `contentType` are checked by the if just above

File: apps/backend-functions/src/templates/watermark.ts (hunk 4 above)
Comment (pldespaigne) [reply apps/backend-functions/src/templates/watermark.ts:5]: Oh sh*t you are right it's global scope

File: apps/backend-functions/src/users.ts (hunk 6 above)
Comment (pldespaigne) [reply apps/backend-functions/src/users.ts:115]: not a big fan of this, it's not easily understandable, and moreover if the `firstName` change but not the `avatar` we will not update algolia

Comment (GrandSchtroumpf) [reply apps/backend-functions/src/users.ts:114]: Ah sorry I wrote too quickly : 
```typescript
const changed = (keys: string[]) => keys.some(key => before[key] !== after[key]);
const nameChanged = changed['firstName', 'lastName', 'email']
if (nameChanged || before.avatar?.url !== after.avatar?.url) {
  promises.push(storeSearchableUser(after));
}
if (nameChanged) {
  promises.push(upsertWatermark(after));
}
```

**Pull request merged**